contracts/mock/MockStrategyAsync.sol

```diff
@@ -8,7 +8,7 @@ import {MockStrategySync} from "./MockStrategySync.sol";
 contract MockStrategyAsync is MockStrategySync {
     constructor(
         address _vault,
-        IERC20 _underlying,
+        address _underlying,
```

Comment:
should this be happening?

Reply:
then why are you changing it? xD

Reply:
because `BaseStrategy` expects address for the `_underlying`, so it's either this or calling `BaseStrategy` constructor with `address(_underlying)` :slightly_smiling_face:

Reply:
what about making the base strategy expect a ERC20?

Reply:
That sounds like a good suggestion, certainly will clarify the expectations for the BaseStrategy. I will make the changes

Reply:
[changed](https://github.com/lindy-labs/sc_solidity-contracts/pull/183/commits/451ae0e291c053712ff4403fac0eb32b1d5515e0)